build/embed.go

```diff
@@ -0,0 +1,161 @@
+package build
+
+import (
+	"bytes"
+	"fmt"
+	"go/ast"
+	"go/parser"
+	"go/token"
+	"strconv"
+
+	"github.com/visualfc/goembed"
+)
+
+func buildIdent(name string) string {
+	return fmt.Sprintf("__gopherjs_embed_%x__", name)
+}
+
+var embed_head = `package %v
+
+import (
+	"embed"
+	_ "unsafe"
+)
+
+//go:linkname __gopherjs_embed_buildFS__ embed.buildFS
+func __gopherjs_embed_buildFS__(list []struct {
+	name string
+	data string
+	hash [16]byte
+}) (f embed.FS)
+`
+
+// embedFiles generates an additional source file, which initializes all variables in the package with a go:embed directive.
+func embedFiles(bp *PackageData, fset *token.FileSet, files []*ast.File) (*ast.File, error) {
+	if len(bp.EmbedPatternPos) == 0 {
+		return nil, nil
+	}
+
+	ems, err := goembed.CheckEmbed(bp.EmbedPatternPos, fset, files)
+	if err != nil {
+		return nil, err
+	}
+
+	r := goembed.NewResolve()
+	for _, v := range ems {
```

Comment:
nit: For consistency, let's rename `v` to `em`.

Reply:
fixed